comms/src/blazingdb/transport/io/reader_writer.cpp

```diff
@@ -29,42 +29,58 @@ namespace io {
 // numBuffers should be equal to number of threads
 PinnedBufferProvider::PinnedBufferProvider(std::size_t sizeBuffers,
                                            std::size_t numBuffers) {
+  this->bufferSize = sizeBuffers;
   this->buffer_counter = numBuffers;
-  for (int bufferIndex = 0; bufferIndex < numBuffers; bufferIndex++) {
+  this->allocations = std::vector<char *>(1);
+  cudaError_t err = cudaMallocHost((void **)&allocations[0], numBuffers * sizeBuffers * 5);
```

Comment:
If we define `numBuffers` we should use it, not just arbitrarily multiply it by 5. The number coming in is right now hardcoded to 20. We can hardcode that to 100, if we want, but we should take out the ` * 5` here

Reply:
agreed